pkg/util/azureclient/applens/applens_client.go

```diff
@@ -0,0 +1,163 @@
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/gofrs/uuid"
+)
+
+// AppLens client is used to interact with the Azure AppLens service.
+type Client struct {
+	endpoint string
+	pipeline runtime.Pipeline
+}
+
+// Endpoint used to create the client.
+func (c *Client) Endpoint() string {
+	return c.endpoint
+}
+
+// NewClient creates a new instance of AppLens client with Azure AD access token authentication. It uses the default pipeline configuration.
+// endpoint - The applens service endpoint to use.
+// cred - The credential used to authenticate with the applens service.
+// options - Optional AppLens client options.  Pass nil to accept default values.
+func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (*Client, error) {
+	scope, err := createScopeFromEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{endpoint: endpoint, pipeline: newPipeline([]policy.Policy{runtime.NewBearerTokenPolicy(cred, scope, nil), &appLensBearerTokenPolicy{}}, o)}, nil
+}
+
+func newPipeline(authPolicy []policy.Policy, options *ClientOptions) runtime.Pipeline {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	return runtime.NewPipeline("applens", serviceLibVersion,
+		runtime.PipelineOptions{
+			PerCall:  []policy.Policy{},
+			PerRetry: authPolicy,
+		},
+		&options.ClientOptions)
+}
+
+func createScopeFromEndpoint(endpoint string) ([]string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return []string{fmt.Sprintf("%s://%s/.default", u.Scheme, u.Hostname())}, nil
+}
+
+//  ListDetectors obtains the list of detectors for a service from AppLens.
+// ctx - The context for the request.
+// o - Options for Read operation.
+func (c *Client) ListDetectors(
+	ctx context.Context,
+	o *ListDetectorsOptions) (*http.Response, error) {
+	if o == nil {
+		o = &ListDetectorsOptions{}
+	}
+
+	azResponse, err := c.sendGetRequest(
+		ctx,
+		o,
+		nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return azResponse, nil
+}
+
+// GetDetector obtains detector information from AppLens.
+// ctx - The context for the request.
+// o - Options for Read operation.
+func (c *Client) GetDetector(
+	ctx context.Context,
+	o *GetDetectorOptions) (*http.Response, error) {
+	if o == nil {
+		o = &GetDetectorOptions{}
+	}
+
+	azResponse, err := c.sendGetRequest(
+		ctx,
+		o,
+		nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return azResponse, nil
+}
+
+func (c *Client) sendGetRequest(
+	ctx context.Context,
+	requestOptions appLensRequestOptions,
+	requestEnricher func(*policy.Request)) (*http.Response, error) {
+	req, err := c.createRequest(ctx, http.MethodGet, requestOptions, requestEnricher)
```

Comment:
AppLens api is Post, iirc

Reply:
Client updated to use Post. Unit tests also updated.